House3D/house.py

```diff
@@ -577,14 +576,42 @@ def is_door(obj):
             if gen_debug_map and (self._debugMap is not None):
                 fill_region(self._debugMap, x1, y1, x2, y2, 0.8)
 
+
     def genMovableMap(self):
-        for i in range(self.n_row+1):
-            for j in range(self.n_row+1):
+        roi_bounds = self._getRegionsOfInterest()
+        for roi in roi_bounds:
+            self._updateMovableMap(*roi)
+
+
+    def _updateMovableMap(self, x1, y1, x2, y2):
+        for i in range(x1, x2):
+            for j in range(y1, y2):
                 if self.obsMap[i,j] == 0:
                     cx, cy = self.to_coor(i, j, True)
                     if self.check_occupy(cx,cy):
                         self.moveMap[i,j] = 1
 
+
+    def _getRegionsOfInterest(self):
+        """Override this function for customizing the areas of the map to
+        consider when marking valid movable locations
+        Returns a list of (x1, y1, x2, y2) tuples representing bounding boxes
+        of valid areas.  Coordinates are normalized grid coordinates.
+        """
+        result = []
+        if restrictToTargets:
```

Comment:
This variable was declared nowhere? I guess this branch can just be removed?

Reply:
Oh weird, I think I forgot to include some change to this file - this function should only have this line:
```
return [(0, 0, self.n_row+1, self.n_row+1)]
```

And then the comment is valid for having overridable functionality here